src/Symfony/Component/Yaml/Inline.php

```diff
@@ -156,12 +156,22 @@ public static function dump($value, $exceptionOnInvalidType = false, $objectSupp
      */
     private static function dumpArray($value, $exceptionOnInvalidType, $objectSupport)
     {
+        if ($value) {
```

Comment:
In the `Dumper` class we use `array_keys($input) !== range(0, count($input) - 1)` to detect the type of array. I am fine with changing the way we detect it, but then we should move the detection into a dedicated method an reuse it in both places.